sound/soc/sof/ops.h

```diff
@@ -470,6 +476,32 @@ snd_sof_set_mach_params(const struct snd_soc_acpi_mach *mach,
 		sof_ops(sdev)->set_mach_params(mach, dev);
 }
 
+static inline void
+snd_sof_register_clients(struct snd_sof_dev *sdev)
+{
+	if (sof_ops(sdev) && sof_ops(sdev)->register_clients)
+		sof_ops(sdev)->register_clients(sdev);
+}
+
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_CLIENT)
+static inline void
+snd_sof_unregister_clients(struct snd_sof_dev *sdev)
+{
+	struct sof_client_dev *cdev, *_cdev;
+
+	/* unregister client devices */
+	mutex_lock(&sdev->client_mutex);
+	list_for_each_entry_safe(cdev, _cdev, &sdev->client_list, list) {
+		list_del(&cdev->list);
+		virtbus_unregister_device(&cdev->vdev);
```

Comment:
nit-pick: do we want to have a sof_client_dev_unregister, with the virtbus dependency handled internally? I feel this really doesn't below in the SOF ops header.

Reply:
hmm sure I can change it